### ISSUE TITLE
libnet-amazon-perl: Net::Amazon broken with Amazon API changes?

### ISSUE THREAD

**Issue by gregoa** [boumenot/p5-Net-Amazon#9]
We have the following bug reported to the Debian package of
Net-Amazon, c.f. https://bugs.debian.org/990042

It doesn't seem to be a bug in the packaging, so you may want to take
a look. Thanks!

------8<-----------8<-----------8<-----------8<-----------8<-----

```
Package: libnet-amazon-perl
Version: 0.62-1
Severity: grave
Tags: upstream
Justification: renders package unusable

I have a script which uses Net::Amazon, and which I run once every
other year. The last attempt ended with no search results but just an
error message of "Gone".

The same can be seen when entering the URL from t/024signature.t in a
browser:
https://webservices.amazon.com/onca/xml?Service=AWSECommerceService&AWSAccessKeyId=YOUR_AMZN_TOKEN&Operation=ItemSearch&Keywords=Bubl%C3%A9&SearchIndex=Music&ResponseGroup=ItemAttributes,Offers&Version=2009-03-31&Timestamp=2009-06-02T16:31:39Z

leads to a status of "410 Gone" and a content saying

"Gone

The requested resource
/onca/xml
is no longer available on this server and there is no forwarding address. Please remove all references to this resource."


A very brief web search indicates that this API endpoint is retired.
Which would make Net::Amazon disfunctional â¦


Notes:
- tested with 0.62-2.1 but this most probably applies back to 0.62-1
- I'll forward the bug upstream once I get a bug number
  Currently there's no issue yet at
  https://rt.cpan.org/Public/Dist/Display.html?Name=Net-Amazon
  or
  https://github.com/boumenot/p5-Net-Amazon/issues
- If Net::Amazon is indeed disfunctional we either need a fix quickly
  or it shouldn't be in the bullseye release
- libnet-amazon-perl has no reverse dependencies
- popcon: vote: 3 + old: 509
- I think it was my first Debian package of a Perl module, back in 2006 :)
- I'll look into XML::Amazon soon


Cheers,
gregor


```
------8<-----------8<-----------8<-----------8<-----------8<-----

Thanks for considering,
  gregor herrmann,
  Debian Perl Group

**Comment by boumenot** [comment]
Thank you for bringing this to my attention.  I haven't touch Net::Amazon in quite some time, and I was unaware of this issue.  Your the first person to open an issue on this fact. Agree, my [search](https://forums.aws.amazon.com/thread.jspa?threadID=322710) tells me that this API was removed in early 2020, and this is the first issue opened about it. ☹️

> 2006 - impressive!

Amazon provides a [new API](https://webservices.amazon.com/paapi5/documentation/).  I do not see an SDK for Perl.

This is a significant amount of work to port Net::Amazon to the new API, and I am not willing to take that work on.  The best recourse is to remove Net::Amazon because it no longer works.

**Comment by gregoa** [comment]
Thanks for the quick reply!

I was suprised as well that no-one has submitted the problems before; apparently Net::Amazon has a small userbase …

I totally understand that porting the module to the new API is too much work, and I agree on your conclusion to retire it.

Thanks for your past work on this module!

**Comment by davorg** [comment]
@boumenot Would you consider [marking the module as ADOPTME](http://blogs.perl.org/users/brian_d_foy/2013/02/mark-your-modules-as-adoptable-if-you-dont-want-them.html) so that someone else could pick it up and do the rewrite?

And, until that happens, perhaps you could release another version to CPAN with the documentation updated to indicate that the module currently doesn't work at all.

**Comment by gregoa** [comment]
On Wed, 29 Sep 2021 08:30:03 -0700, Dave Cross wrote:

> @boumenot Would you consider [marking the module as ADOPTME](http://blogs.perl.org/users/brian_d_foy/2013/02/mark-your-modules-as-adoptable-if-you-dont-want-them.html) so that someone else could pick it up and do the rewrite?
> 
> And, until that happens, perhaps you could release another version to CPAN with the documentation updated to indicate that the module currently doesn't work at all.

BTW, this affect XML-Amazon as well:

https://github.com/shlomif/perl-XML-Amazon/issues/1

Cheers,
gregor

-- 
 .''`.  https://info.comodo.priv.at -- Debian Developer https://www.debian.org
 : :' : OpenPGP fingerprint D1E1 316E 93A7 60A8 104D  85FA BB3A 6801 8649 AA06
 `. `'  Member VIBE!AT & SPI Inc. -- Supporter Free Software Foundation Europe
   `-

**Comment by davorg** [comment]
@gregoa Yes, I've just found that out too. Looks like I'm following the same path as you did - just three months later :-)
